src/components/select/filterable-select/filterable-select.component.tsx

```diff
@@ -211,8 +211,8 @@ export const FilterableSelect = React.forwardRef(
       textToMatch: string,
       list: React.ReactNode
     ) {
-      return (list as React.ReactElement[]).find((child) => {
-        const { text } = child.props;
+      return React.Children.toArray(list).find((child) => {
+        const { text } = (child as React.ReactElement).props;
 
         return text?.toLowerCase().indexOf(textToMatch?.toLowerCase()) !== -1;
```

Comment:
**suggestion**: what do you think about amending the typing like this, to avoid having to use type assertions and mask potential type errors?:

```suggestion
      return React.Children.toArray(list).find(
        (child): child is React.ReactElement<OptionProps> => {
          if (!React.isValidElement<OptionProps>) return;
          
          const { text } = child.props;

          return text.toLowerCase().indexOf(textToMatch?.toLowerCase()) !== -1;
```

Reply:
It should work if I pass the `child` as an argument to `isValidElement`.